src/features/diagnosticsProvider.ts

```diff
@@ -136,26 +136,24 @@ class DiagnosticsProvider extends AbstractSupport {
         this._suppressHiddenDiagnostics = vscode.workspace.getConfiguration('csharp').get('suppressHiddenDiagnostics', true);
 
         this._subscriptions.push(this._validateCurrentDocumentPipe
-            .asObservable()
-            .pipe(throttleTime(750))
-            .subscribe(async x => await this._validateDocument(x)));
+            .pipe(debounceTime(750))
+            .subscribe(x => this._validateDocument(x)));
 
         this._subscriptions.push(this._validateAllPipe
-            .asObservable()
-            .pipe(throttleTime(3000))
-            .subscribe(async () => {
+            .pipe(debounceTime(3000))
```

Comment:
throttle and debounce work similarly. Throttle fires on first request then ignores requests for a set amount of time. Debounce will wait for incoming requests to settle down before firing. This is an important distinction because we would want to wait for a users to be finished editing a document before requesting diagnostics.